app/code/Magento/Indexer/Console/Command/IndexerStatusCommand.php

```diff
@@ -30,21 +32,84 @@ protected function configure()
+        $table->setHeaders(['Title', 'Status', 'Update On', 'Schedule Status', 'Schedule Updated']);
+
+        $rows = [];
+
         $indexers = $this->getIndexers($input);
         foreach ($indexers as $indexer) {
-            $status = 'unknown';
-            switch ($indexer->getStatus()) {
-                case \Magento\Framework\Indexer\StateInterface::STATUS_VALID:
-                    $status = 'Ready';
-                    break;
-                case \Magento\Framework\Indexer\StateInterface::STATUS_INVALID:
-                    $status = 'Reindex required';
-                    break;
-                case \Magento\Framework\Indexer\StateInterface::STATUS_WORKING:
-                    $status = 'Processing';
-                    break;
+            $view = $indexer->getView();
+
+            $rowData = [
+                'Title'             => $indexer->getTitle(),
+                'Status'            => $this->getStatus($indexer),
+                'Update On'         => $indexer->isScheduled() ? 'Schedule' : 'Save',
+                'Schedule Status'   => '',
+                'Updated'           => '',
+            ];
+
+            if ($indexer->isScheduled()) {
+                $state = $view->getState();
+                $rowData['Schedule Status'] = "{$state->getStatus()} ({$this->getPendingCount($view)} in backlog)";
+                $rowData['Updated'] = $state->getUpdated();
             }
-            $output->writeln(sprintf('%-50s ', $indexer->getTitle() . ':') . $status);
+
+            $rows[] = $rowData;
+        }
+
+        usort($rows, function ($comp1, $comp2) {
+            return strcmp($comp1['Title'], $comp2['Title']);
+        });
+
+        $table->addRows($rows);
+        $table->render($output);
+    }
+
+    /**
+     * @param Indexer\IndexerInterface $indexer
+     * @return string
+     */
+    private function getStatus(Indexer\IndexerInterface $indexer)
+    {
+        $status = 'unknown';
+        switch ($indexer->getStatus()) {
+            case \Magento\Framework\Indexer\StateInterface::STATUS_VALID:
+                $status = 'Ready';
+                break;
+            case \Magento\Framework\Indexer\StateInterface::STATUS_INVALID:
+                $status = 'Reindex required';
+                break;
+            case \Magento\Framework\Indexer\StateInterface::STATUS_WORKING:
+                $status = 'Processing';
+                break;
         }
+        return $status;
+    }
+
+    /**
+     * @param Mview\ViewInterface $view
+     * @return string
+     */
+    private function getPendingCount(Mview\ViewInterface $view)
+    {
+        $changelog = $view->getChangelog();
+
+        try {
+            $currentVersionId = $changelog->getVersion();
+        } catch (Mview\View\ChangelogTableNotExistsException $e) {
+            return '';
+        }
+
+        $state = $view->getState();
+
+        $pendingCount = $changelog->getListSize($state->getVersionId(), $currentVersionId);
```

Comment:
`\Magento\Framework\Mview\ViewInterface::getChangelog` returns `\Magento\Framework\Mview\View\ChangelogInterface` so method `getListSize` cannot be used here as it is declared in `ChangelogCounterInterface` and is unkown in this code.

Seems there is no *clean* and *backward compatible* solution to implement counting without list load. I propose in the scope of this PR implement counting as was done in the initial version with `count($changelog->getList())` and process this PR to deliver this feature to M2. Once PR will be merged new PR may be created with a refactoring of changelog and performance optimization. 

@convenient, @ihor-sviziev, @kandy are you agree with this plan?

Reply:
Thanks @vkublytskyi I was really struggling to balance out clean/backwards-compatability as you noted. 

If @kandy agrees my plan of action will be
1. Roll back to `count($changelog->getList())` with associated tests
2. Update description/title to properly highlight what is occurring in this PR
3. Wait for approval from all parties (or potentially for merge?)
4. Backport changes to #12050 

Sound good?

Reply:
Ok, looks good for me